test/e2e/subscription_e2e_test.go

```diff
@@ -1287,3 +1310,103 @@ func updateInternalCatalog(t *testing.T, c operatorclient.ClientInterface, crc v
 	})
 	require.NoError(t, err)
 }
+
+// TestSingleInstallPlanCreatedPerSubscription ensures that applying N number of subscriptions only creates
+// N number of InstallPlans.
+//
+// Steps:
+// 1. Create namespace, ns
+// 2. Create CatalogSource, cs, in ns
+// 3. Create multiple Subscriptions to each package of cs in ns, sub
+// 4. Wait for the package from sub to install successfully with no remaining InstallPlan status conditions
+// 5. Ensure that the InstallPlan is correctly referenced in the subscription
+// 6. Ensure that the number of InstallPlans created matches the number of subs created
+func TestSingleInstallPlanCreatedPerSubscription(t *testing.T) {
+	defer cleaner.NotifyTestComplete(t, true)
+
+	c := newKubeClient(t)
+	crc := newCRClient(t)
+
+	// Create namespace ns
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: genName("ns-"),
+		},
+	}
+	_, err := c.KubernetesInterface().CoreV1().Namespaces().Create(ns)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, c.KubernetesInterface().CoreV1().Namespaces().Delete(ns.GetName(), &metav1.DeleteOptions{}))
+	}()
+
+	Numpkgs := 2
+
+	// Create a CatalogSource, cs, in ns with Numpkgs pkgs
+	pkgNames := []string{}
+	channelNames := []string{}
+	crds := []apiextensions.CustomResourceDefinition{}
```

Comment:
this seems like a good helper function to pull out and share with other tests (this section generating catalogs)